packages/bvaughn-architecture-demo/src/suspense/GenericCache.ts

```diff
@@ -0,0 +1,54 @@
+import { Record, STATUS_PENDING, STATUS_REJECTED, STATUS_RESOLVED, Wakeable } from "./types";
+import { createWakeable } from "../utils/suspense";
+
+export function createGenericCache<TParams extends Array<any>, TValue>(
+  fetchValue: (...args: TParams) => Promise<TValue>,
+  getCacheKey: (...args: TParams) => string
+) {
+  const recordMap = new Map<string, Record<TValue>>();
+
+  async function fetchAndStoreValue(
+    record: Record<TValue>,
+    wakeable: Wakeable<TValue>,
+    ...args: TParams
+  ) {
+    try {
+      const value = await fetchValue(...args);
+
+      record.status = STATUS_RESOLVED;
+      record.value = value;
+
+      wakeable.resolve(record.value);
```

Comment:
Super tiny nit: You have an unnecessary property access here (`record.value`) since `value` is in scope right above.